File: lib/child_workflow/child_workflow_child_workflow.ex
defmodule ChildWorkflow.ChildWorkflow do
  use TemporalSdk.Workflow
  @moduledoc false

  @impl true

  def execute(_context, [["cancel"]]),
    do: cancel_workflow_execution([["Test child WF cancellation."]])

  def execute(_context, [["fail"]]),
    do: fail_workflow_execution(message: "Test child WF failure.")

  def execute(_context, [["invalid"]]),
    do: complete_workflow_execution([12345])

  def execute(_context, _input) do
    %{result: r} = start_activity(EchoActivity, [["from temporal"]], [:wait])
    complete_workflow_execution(r)
  end
end

File: src/child_workflow/child_workflow_child_workflow.erl
-module(child_workflow_child_workflow).

% elp:ignore W0012 W0040
-moduledoc false.

-export([
    execute/2
]).

-include_lib("temporal_sdk/include/workflow.hrl").

execute(_Context, [[~"cancel"]]) ->
    cancel_workflow_execution([[~"Test child WF cancellation."]]);
execute(_Context, [[~"fail"]]) ->
    fail_workflow_execution([{message, "Test child WF failure."}]);
execute(_Context, [[~"invalid"]]) ->
    complete_workflow_execution([12345]);
execute(_Context, _Input) ->
    #{result := R} = start_activity(echo_activity, [[~"from temporal"]], [wait]),
    complete_workflow_execution(R).
